api/envoy/config/bootstrap/v2/bootstrap.proto

```diff
@@ -228,3 +234,55 @@ message Runtime {
   // encoding <config_runtime_proto_json>`.
   google.protobuf.Struct base = 4;
 }
+
+message RuntimeLayer {
+  // :ref:`Static runtime <config_runtime_bootstrap>` layer.
+  message StaticLayer {
+    // This follows the :ref:`runtime protobuf JSON representation encoding
+    // <config_runtime_proto_json>`.
+    google.protobuf.Struct value = 1;
+  }
+
+  // :ref:`Disk runtime <config_runtime_local_disk>` layer.
+  message DiskLayer {
+    // Specifies the subdirectory to load within the :ref:`root directory
+    // <envoy_api_field_config.bootstrap.v2.LayeredRuntime.symlink_root>`. This
+    // is useful if multiple systems share the same delivery mechanism. Envoy
+    // configuration elements can be contained in a dedicated subdirectory.
+    string subdirectory = 1;
+
+    // :ref:`Append <config_runtime_local_disk_overrides>` the service cluster
+    // to the path.
+    bool append_service_cluster = 2;
+  }
+
+  // :ref:`Admin console runtime <config_runtime_admin>` layer.
+  message AdminLayer {
+  }
+
+  // Descriptive name for the runtime layer. This is only used for the runtime
+  // :http:get:`/runtime` output.
+  string name = 1;
+
+  oneof layer_specifier {
+    google.protobuf.Struct static_layer = 2;
+    DiskLayer disk_layer = 3;
+    AdminLayer admin_layer = 4;
+  }
+}
+
+// Runtime :ref:`configuration overview <config_runtime>`.
+message LayeredRuntime {
+  // For disk-based runtime layers, the implementation assumes that the file
```

Comment:
Since we are redoing this API, WDYT about making the symlink_root and watch be part of each DiskLayer? I don't see any particular reason to have it be shared?

Reply:
I'm good either way, let me know your preference. It's marginally easier implementation wise to have one watcher rather than N, but that's marginal. I had thought the reason for a common symlink root was to allow all disk layers to be atomically swapped; if we have multiple symlink roots, we will need to make sure that when they share a root, this atomic property remains. That's probably as simple as deduping the watchers. WDYT?

Reply:
I guess my feeling is that it's cleaner from an API perspective, and given that we now have N layers which can come in at different times, I'm not sure that the whole thing is atomic anyway, so it probably can be separated. If you are up for it, I would move it, but I don't feel super strongly about it.

Reply:
Some examples for discussion. Imagine the following scenarios:
1. We have two DiskLayers, with symlink /a and /a. How many reloads of the filesystem should we do when /a or anything inside /a changes? Presumably 1, this is easy to dedupe at watcher instantiation.
2. We have two DiskLayers, with symlink /a and /a/b. How many reloads of the filesystem should we do when /a/b or anything inside /a/b changes? Presumably 1, but this requires some non-trivial reasoning about relative paths.
3. We have two DiskLayers, with symlink /a and /b. I think it's clear here what should happen.

So, yeah, just wondering what are the expected semantics around (2)?

Reply:
I think I'm going to leave (2) as "use at your own risk, you might get multiple reloads".

Reply:
IMO that's fine. I don't think multiple reloads really matters.